### ISSUE TITLE
When using non-stream mode, the client is blocking.

### ISSUE THREAD

**Issue by wzzju** [EricLBuehler/candle-vllm#75]
When I run the Yi-6B model inference service using the command `cargo run --release -- --port 2000 --weight-path /home/yi-6b/ yi --repeat-last-n 32` and then execute `python examples/llama.py` as a client test, the client program keeps blocking. After pressing Ctrl+C, the following output can be seen on the server:

```
Prompt "<|im_start|>user\n Explain how to best learn Rust. <|im_end|>"
Request cmpl-8f264377-117d-4e63-87b3-41621869c188 with length 13 added to sequence group.
Send stream response error!
Request cmpl-8f264377-117d-4e63-87b3-41621869c188 decoding finished in 0 seconds

 [1 requests] Prefilling: 13 prompt tokens processed in 289 seconds

 [1 requests] Decoding: 0 tokens processed in 0 seconds (0 tokens/s)
```

However, running `python examples/benchmark.py` works successfully.  The following curl test command also gets stuck.

```sh
curl -X POST "http://127.0.0.1:2000/v1/chat/completions" \
     -H "Content-Type: application/json" \
     -H "Authorization: Bearer YOUR_API_KEY" \
     -d '{
           "model": "llama7b",
           "messages": [
               {"role": "user", "content": "Explain how to best learn Rust."}
           ],
           "temperature": 0.7,
          "max_tokens": 128,
          "stop": {"Single":"</s>"}
       }'
```

**Comment by guoqingbao** [comment]
> When I run the Yi-6B model inference service using the command `cargo run --release -- --port 2000 --weight-path /home/yi-6b/ yi --repeat-last-n 32` and then execute `python examples/llama.py` as a client test, the client program keeps blocking. After pressing Ctrl+C, the following output can be seen on the server:
> 
> ```
> Prompt "<|im_start|>user\n Explain how to best learn Rust. <|im_end|>"
> Request cmpl-8f264377-117d-4e63-87b3-41621869c188 with length 13 added to sequence group.
> Send stream response error!
> Request cmpl-8f264377-117d-4e63-87b3-41621869c188 decoding finished in 0 seconds
> 
>  [1 requests] Prefilling: 13 prompt tokens processed in 289 seconds
> 
>  [1 requests] Decoding: 0 tokens processed in 0 seconds (0 tokens/s)
> ```
> 
> However, running `python examples/benchmark.py` works successfully. The following curl test command also gets stuck.
> 
> ```shell
> curl -X POST "http://127.0.0.1:2000/v1/chat/completions" \
>      -H "Content-Type: application/json" \
>      -H "Authorization: Bearer YOUR_API_KEY" \
>      -d '{
>            "model": "llama7b",
>            "messages": [
>                {"role": "user", "content": "Explain how to best learn Rust."}
>            ],
>            "temperature": 0.7,
>           "max_tokens": 128,
>           "stop": {"Single":"</s>"}
>        }'
> ```

Which type of GPU device you have used? The non-stream mode is designed to block the caller, I'm not sure if it blocks the calling thread without any response in the end (bug) or it just takes a long time for response (depend on the GPU capability).

**Comment by guoqingbao** [comment]
This issue has been addressed in #80 , please use the latest update.
